sqlalchemy_ibmi/base.py

```diff
@@ -675,7 +688,12 @@ class IBMiDb2Dialect(default.DefaultDialect):
     colspecs = COLSPECS
     ischema_names = ISCHEMA_NAMES
     supports_unicode_binds = True
-    returns_unicode_strings = False
+    if SA_Version < [1, 4]:
+        returns_unicode_strings = False
+    elif SA_Version < [2, 0]:
+        returns_unicode_strings = sa_types.String.RETURNS_CONDITIONAL
+    else:
+        returns_unicode_strings = True
```

Comment:
I don't think this complication is needed, based on the description here: https://docs.sqlalchemy.org/en/14/changelog/changelog_14.html#change-d3650b27f4c852ac687ea55326051ed0

As we don't support Python 2 and on Python 3, pyodbc returns strings for character types, I don't think there's any condition where we would return bytes for character data.

```suggestion
    returns_unicode_strings = True
```

Reply:
I was having issues with returns_unicode_strings = True in Superset. I tried a few things trying to get it to work but pulling this snippet from ibm_db_sa seemed to work for me.